ios/Classes/NativeBridge/Handlers/MobileAuthHandler.swift

```diff
@@ -2,132 +2,128 @@ import Foundation
 import OneginiSDKiOS
 
 protocol MobileAuthConnectorToHandlerProtocol: AnyObject {
-    func enrollForMobileAuth(_ completion: @escaping (Bool?, SdkError?) -> Void)
-    func isUserEnrolledForMobileAuth() -> Bool
-    func handleMobileAuthConfirmation(cancelled: Bool)
-    func handleOTPMobileAuth(_ otp: String , customRegistrationChallenge: ONGCustomRegistrationChallenge?, _ completion: @escaping (Any?, SdkError?) -> Void)
-    func handleQrOTPMobileAuth(_ otp: String , customRegistrationChallenge: ONGCustomRegistrationChallenge?, _ completion: @escaping (Any?, SdkError?) -> Void)
-}
-
-enum MobileAuthAuthenticatorType: String {
-    case fingerprint = "biometric"
-    case pin = "PIN"
-    case confirmation = ""
+    func handleMobileAuthWithOtp(otp: String, completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func enrollMobileAuthentication(completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func acceptMobileAuthRequest(completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func denyMobileAuthRequest(completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func setMobileAuthCallback(didReceiveConfirmationChallenge confirmation: @escaping (Bool) -> Void)
 }
 
 class MobileAuthHandler: NSObject {
-    var authenticatorType: MobileAuthAuthenticatorType?
-    var confirmation: ((Bool) -> Void)?
-    var mobileAuthCompletion: ((Any?, SdkError?) -> Void)?
-        
-    fileprivate func handleConfirmationMobileAuth(_ cancelled: Bool) {
-        guard let confirmation = confirmation else { fatalError() }
-        
-        confirmation(cancelled)
-    }
+    var mobileAuthCallback: ((Bool) -> Void)?
+    var flowStarted: Bool = false
 }
 
 //MARK: - MobileAuthConnectorToHandlerProtocol
 extension MobileAuthHandler : MobileAuthConnectorToHandlerProtocol {
-    func enrollForMobileAuth(_ completion: @escaping (Bool?, SdkError?) -> Void) {
-        ONGClient.sharedInstance().userClient.enroll { enrolled, error in
-            if let error = error {
-                let mappedError = ErrorMapper().mapError(error);
-                completion(false, mappedError)
-              } else {
-                if(enrolled == false){
-                    completion(false, SdkError(.enrollmentFailed))
-                    return;
-                }
-                
-                completion(true, nil)
-              }
+    func handleMobileAuthWithOtp(otp: String, completion: @escaping (Result<Void, FlutterError>) -> Void) {
+        Logger.log("handleMobileAuthWithOtp", sender: self)
+
+        // Check to prevent breaking iOS SDK; https://onewelcome.atlassian.net/browse/SDKIOS-987
+        guard let _ = SharedUserClient.instance.authenticatedUserProfile else {
```

Comment:
if `SharedUserClient.instance.authenticatedUserProfile` is marked as `@discardableResult` you don't need `_ =` or you can check 
```
guard SharedUserClient.instance.authenticatedUserProfile == nil else {
...

Reply:
done